src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/ItemMoveListener.java

```diff
@@ -0,0 +1,32 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunItems;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.api.BlockStorage;
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryMoveItemEvent;
+
+import javax.annotation.Nonnull;
+
+import java.util.logging.Level;
+
+public class ItemMoveListener implements Listener {
+
+    public ItemMoveListener(@Nonnull SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onItemMove(InventoryMoveItemEvent e) {
+        Location loc = e.getDestination().getLocation();
+        if (loc != null && BlockStorage.hasBlockInfo(loc)) {
+            if (BlockStorage.check(loc, SlimefunItems.ELECTRIC_ORE_GRINDER.getItemId())) {
```

Comment:
not sure if this adds all furnace type machines but for slimefun it self you forgot Electric furnace and Electric Smeltery
it would be nice if this blocks it for all Furnace type machines even for addons.

Reply:
Yes, I will be adding the rest and excluding the Enchanced Furnaces. I don't know how to though without making it messy. Will look through the codebase to get an idea.